internal/yggdrasil/yggdrasil.go

```diff
@@ -357,6 +364,12 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 			h.metrics.IncEdgeDeviceFailedRegistration()
 			return operations.NewPostDataMessageForDeviceInternalServerError()
 		}
+		err = h.deviceRepository.UpdateLabels(ctx, &device, hardware.MapLabels(registrationInfo.Hardware))
```

Comment:
this shouldn't produce too much load (expected to be called once) and yet, could update the labels as part of the device creation or is the order(create device --> update status --> update labels) important here?

Reply:
From my point of view the order is not important. Do you think we should change it?

Reply:
I think it should be in reconcile. Besides status (that has to be here, sort of) we do all such operations in reconcile. 
A client sends a request and should get a response asap. Less load on HTTP server. Think of many heartbeats.
Another motivation is that every step s another possible failure for the registration/heartbeat. I don't think it should fail it.

Reply:
@pkliczewski see my comment to this thread

Reply:
when heartbeat patches the status 'EdgeDeviceReconciler.Reconcile' is called. you can verify there that the hardware labels match the hardware status. For registration, it is not patch. it is create. still, it gets to Reconcile.